File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int todaysProfit=0, totalProfit=0, buyDay=0;
        int n=prices.size()-1;

        for(int i=1; i<=n; i++) {
            todaysProfit = prices[i] - prices[buyDay];
            if(todaysProfit<=0) {
                buyDay=i;
            }
            if(todaysProfit > totalProfit) {
                totalProfit = todaysProfit;
            }
        }
        return totalProfit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {

        int n = prices.length;
        int minPrice = Integer.MAX_VALUE;
        int totalProfit = 0;
       
        for(int i=0; i<n; i++) {
            minPrice = Math.min(minPrice, prices[i]);

            int profit = prices[i] - minPrice;

            totalProfit = Math.max(totalProfit, profit);
        }
        return totalProfit;
    }
}